src/features/auth/projectRedactorSignup/SetRedactorEmail.tsx

```diff
@@ -0,0 +1,114 @@
+import { QuitSignupModal, SignupSteps } from 'features/auth/components/QuitSignupModal'
+import { SetEmailModalContent, StyledInput } from 'features/auth/components/signupComponents'
+import { useBackNavigation } from 'features/navigation/backNavigation'
+import { UseNavigationType } from 'features/navigation/RootNavigator'
+import { testID } from 'tests/utils'
+import { BottomContentPage } from 'ui/components/BottomContentPage'
+import { ButtonPrimary } from 'ui/components/buttons/ButtonPrimary'
+import { isEmailValid } from 'ui/components/inputs/emailCheck'
+import { isValueEmpty } from 'ui/components/inputs/helpers'
+import { InputError } from 'ui/components/inputs/InputError'
+import { TextInput } from 'ui/components/inputs/TextInput'
+import { ModalHeader } from 'ui/components/modals/ModalHeader'
+import { useModal } from 'ui/components/modals/useModal'
+import { StepDots } from 'ui/components/StepDots'
+import { ArrowPrevious } from 'ui/svg/icons/ArrowPrevious'
+import { Close } from 'ui/svg/icons/Close'
+import { Spacer, Typo } from 'ui/theme'
+
+export const SetRedactorEmail: FunctionComponent = () => {
+  const [email, setEmail] = useState('')
+  const [hasError, setHasError] = useState(false)
+
+  const navigation = useNavigation<UseNavigationType>()
+
+  const shouldDisableValidateButton = isValueEmpty(email)
+
+  const emailInput = useRef<RNTextInput | null>(null)
+
+  const {
+    visible: fullPageModalVisible,
+    showModal: showFullPageModal,
+    hideModal: hideFullPageModal,
+  } = useModal(false)
+
+  const complexGoBack = useBackNavigation()
+  const numberOfSteps = useRedactorSignUpNumberOfSteps()
+
+  function onChangeEmail(email: string) {
+    if (hasError) {
+      setHasError(false)
+    }
+    setEmail(email)
+  }
+
+  async function validateEmail() {
+    if (isEmailValid(email)) {
+      navigation.navigate(`SetRedactorPassword`, { email })
+    } else {
+      setHasError(true)
+    }
+  }
+
+  function showQuitSignupModal() {
+    emailInput.current && emailInput.current.blur()
+    showFullPageModal()
+  }
+
+  return (
+    <React.Fragment>
+      <BottomContentPage>
+        <ModalHeader
+          title={t`Votre e-mail`}
+          leftIcon={ArrowPrevious}
+          onLeftIconPress={complexGoBack}
+          rightIcon={Close}
+          onRightIconPress={showQuitSignupModal}
+        />
+        <SetEmailModalContent>
+          <StyledInput>
+            <Typo.Body>{t`Adresse e-mail`}</Typo.Body>
+            <Spacer.Column numberOfSpaces={2} />
+            <TextInput
+              autoCapitalize="none"
+              autoFocus={true}
+              keyboardType="email-address"
+              onChangeText={onChangeEmail}
+              placeholder={t`votreadresse@email.com`}
+              ref={emailInput}
+              textContentType="emailAddress"
+              value={email}
+              {...testID("Entrée pour l'email")}
+            />
+            <InputError
+              visible={hasError}
```

Comment:
je sais pas si on veut garder la même UX, mais on peut se passer du `useState` et passer directement:

```suggestion
              visible={!isEmailValid(email)}
```

comme ça on a un feedback à chaque instant, pas quand on clique sur "Continuer"

Reply:
à condition que isEmailValid soit optimisée

Reply:
`à condition que isEmailValid soit optimisée` c'est à dire ?
Pour le feedback à chaque instant je suis pas convaincue, ça voudrait dire que l'utilisateur voit un message d'erreur "format de l'email incorrect" jusqu'à ce qu'il ait finit d'entrer son email 🤔

Reply:
okey comme ça